### DISCUSSION TITLE
AzureADB2C  Provider - is this ADAL or MSAL?

### DISCUSSION THREAD

**Discussion by yuklai** [nextauthjs/next-auth#2484] [Help]
Hi all, 

Does anyone know if this is using ADAL or MSAL? 
https://docs.microsoft.com/en-us/azure/active-directory/develop/msal-migration

I assume that this is ADAL? I can't seem to find a good way to confirm. Would love to learn more if anyone can shed some lights. 

If this is ADAL, does anyone know of a MSAL implementation? There are some differences in the way refreshToken is managed.

Thanks!

**Comment by dislick** [comment]
Hi, I've been wondering that as well. Have you found out yet @yuklai?
